src/burst_db/_land_usgs.py

```diff
@@ -0,0 +1,108 @@
+"""An internal module to download shape files for land are and Greenland."""
+import fnmatch
+import zipfile
+from pathlib import Path
+
+import geopandas as gpd
+import pandas as pd
+import requests
+import unzip_http
+from shapely.geometry import MultiPolygon
+
+USGS_LAND_URL = "https://www.ngdc.noaa.gov/mgg/shorelines/data/gshhg/latest/gshhg-shp-2.3.7.zip"  # noqa
+GREENLAND_URL = "https://stacks.stanford.edu/file/druid:sd368wz2435/data.zip"  # noqa
+
+
+def get_usgs_land(outpath=None):
+    """Get the USGS land data from the following url:
+
+    https://www.ngdc.noaa.gov/mgg/shorelines/data/gshhg/latest/gshhg-shp-2.3.7.zip
+    """
+    outpath = Path(outpath) if outpath else Path.cwd()
+    rzf = unzip_http.RemoteZipFile(USGS_LAND_URL)
+    # Level 1: Continental land masses and ocean islands, except Antarctica.
+    # Level 6: Antarctica based on grounding line boundary.
+    paths = ["GSHHS_shp/h/GSHHS_h_L1.*", "GSHHS_shp/h/GSHHS_h_L6.*"]
+    shp_files = []
+    dfs = []
+    for fn in rzf.infolist():
+        if not any(fnmatch.fnmatch(fn.filename, g) for g in paths):
+            continue
+        outname = outpath / fn.filename
+        if outname.suffix == (".shp"):
+            shp_files.append(outname)
+        if not outname.exists():
+            outname.parent.mkdir(parents=True, exist_ok=True)
+            with rzf.open(fn) as fp, open(outname, "wb") as fout:
+                print(f"Extracting {fn.filename} to {outname}")
+                while r := fp.read(2**18):
+                    fout.write(r)
+    for p in shp_files:
+        dfs.append(gpd.read_file(p))
+    return dfs
+
+
+def get_land_df(
+    buffer_deg=0.2,
+    outname="usgs_land_{d}deg_buffered.geojson",
+    driver="GeoJSON",
+    zip=True,
```

Comment:
I would suggest to use another name than `zip` for this parameter, because python has a built-in class with this name.

Reply:
changed to `do_zip`